assets/src/apps/delivery/components/ActivityRenderer.tsx

```diff
@@ -80,8 +82,39 @@ const ActivityRenderer: React.FC<ActivityRendererProps> = ({
   onRequestLatestState = async () => ({ snapshot: {} }),
 }) => {
   const isPreviewMode = useSelector(selectPreviewMode);
-  const currentUserId = 1; // TODO from state
+  const currentUserId = useSelector(selectUserId);
 
+  const saveUserData = async (attemptGuid: string, partAttemptGuid: string, payload: any) => {
+    const objId = `${payload.key}`;
+    debouncedSaveData({ isPreviewMode, payload, objId, value: payload.value });
+  };
+
+  const readUserData = async (attemptGuid: string, partAttemptGuid: string, payload: any) => {
+    // Read only the key from the simid
+    const objId = `${payload.key}`;
+    const data = debouncedReadData({ isPreviewMode, payload, objId });
+    return data;
+  };
+
+  const debouncedReadData = debounce(
+    async ({ isPreviewMode, payload, objId }) => {
+      const retrievedData = await Extrinsic.readGlobalUserState(
+        [`${payload.simId}`],
```

Comment:
`payload.simId` is a string, it doesn't need to be wrapped in a template literal

Reply:
Addressed.